mapie/multi_label_classification.py

```diff
@@ -0,0 +1,746 @@
+        None is a marker for `unset` that will be interpreted as ``n_jobs=1``
+        (sequential execution).
+
+        By default ``None``.
+
+    random_state: Optional[Union[int, RandomState]]
+        Pseudo random number generator state used for random uniform sampling
+        to evaluate quantiles and prediction sets in cumulated_score method.
+        Pass an int for reproducible output across multiple function calls.
+
+        By default ``1``.
+
+    verbose : int, optional
+        The verbosity level, used with joblib for multiprocessing.
+        For the moment, parallel processing is disabled.
+        The frequency of the messages increases with the verbosity level.
+        If it more than ``10``, all iterations are reported.
+        Above ``50``, the output is sent to stdout.
+
+        By default ``0``.
+
+    Attributes
+    ----------
+    valid_methods: List[str]
+        List of all valid methods. Either CRC or RCPS
+
+    single_estimator_ : sklearn.ClassifierMixin
+        Estimator fitted on the whole training set.
+
+    n_lambdas: int
+        Number of thresolds on which we compute the risk.
+
+    risks : ArrayLike of shape (n_samples_cal, n_lambdas)
+        The risk for each observation for each threshold
+
+    r_hat : ArrayLike of shape (n_lambdas)
+        Average risk for each lambda
+
+    r_hat_plus: ArrayLike of shape (n_lambdas)
+        Upper confidence bound for each lambda, computed
+        with different bounds (see predict). Only relevant when
+        method="rcps".
+
+    lambdas_star: ArrayLike of shape (n_lambdas)
+        Optimal threshold according to alpha.
+
+    References
+    ----------
+    [1] Lihua Lei Jitendra Malik Stephen Bates, Anastasios Angelopoulos
+    and Michael I. Jordan. Distribution-free, risk-controlling prediction
+    sets. CoRR, abs/2101.02703, 2021.
+    URL https://arxiv.org/abs/2101.02703.39
+
+    [2] Angelopoulos, Anastasios N., Stephen, Bates, Adam, Fisch, Lihua,
+    Lei, and Tal, Schuster. "Conformal Risk Control." (2022).
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.multioutput import MultiOutputClassifier
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from mapie.multi_label_classification import MapieMultiLabelClassifier
+    >>> X_toy = np.arange(4).reshape(-1, 1)
+    >>> y_toy = np.stack([[1, 0, 1], [1, 0, 0], [0, 1, 1], [0, 1, 0]])
+    >>> clf = MultiOutputClassifier(LogisticRegression()).fit(X_toy, y_toy)
+    >>> mapie = MapieMultiLabelClassifier(estimator=clf).fit(X_toy, y_toy)
+    >>> _, y_pi_mapie = mapie.predict(X_toy, alpha=0.3)
+    >>> print(y_pi_mapie[:, :, 0])
+    [[ True False  True]
+     [ True False  True]
+     [False  True False]
+     [False  True False]]
+    """
+    valid_methods_ = ["crc", "rcps"]
+    n_lambdas = 100
+    fit_attributes = [
+        "single_estimator_",
+        "risks"
+    ]
+    sigma_init = .25  # Value given in the paper.
```

Comment:
Could this not be a input in `fit` or `__init__`?

Reply:
I think it is a parameter one should modify only if he knows the paper very well, that's why I put it as a class attribute as almost nobody will have to change it